src/components/portal/Portal.jsx

```diff
@@ -0,0 +1,73 @@
+import "./portal.css";
+import { useEffect, useRef, useState } from "react";
+import { createPortal } from "react-dom";
+
+const Portal = ({ children, dismiss, anchorRef }) => {
+  const [coordinates, setCoordinates] = useState({});
+
+  const getCoordinates = () => {
+    const rect = anchorRef.current.getBoundingClientRect();
+    const coords = {
+      left: rect.x + rect.width / 2, 
+      top: rect.y, 
+    };
+    return coords;
+  };
```

Comment:
this is something new to me. What portal component is supposed to do ?

Reply:
Render to a complete different DOM element.